### PR TITLE
docs: add descriptions for types of permissions

### LINKED ISSUE #23609: Improve docs for setPermissionRequestHandler
### Preflight Checklist

* [x] I have read the [Contributing Guidelines](https://github.com/electron/electron/blob/master/CONTRIBUTING.md) for this project.
* [x] I agree to follow the [Code of Conduct](https://github.com/electron/electron/blob/master/CODE_OF_CONDUCT.md) that this project adheres to.
* [x] I have searched the issue tracker for an issue that matches the one I want to file, without success.

### Issue Details

* **Electron Version:** 8.2.0

The [docs](https://www.electronjs.org/docs/api/session#sessetpermissionrequesthandlerhandler) list the permissions that we can reject:

>Enum of 'media', 'geolocation', 'notifications', 'midiSysex', 'pointerLock', 'fullscreen', 'openExternal'.

Could the docs be updated to include descriptions of these events? I couldn't find a description of them [here](https://developer.chrome.com/apps/permissions). There's not a one-to-one mapping.

Like what is `openExternal` - is that for new windows?
What media does the `media` permission control exactly?
etc

### PR DESCRIPTION
**Pull request by kritisingh1**
#### Description of Change
Fixes #23609 . Descriptions of various types of permissions are intended included in the documentation via this PR.
@erickzhao Please review.

#### Checklist

- [x] PR description included and stakeholders cc'd
- [x] PR title follows semantic [commit guidelines](https://github.com/electron/electron/blob/master/docs/development/pull-requests.md#commit-message-guidelines)

Notes: none

### CODE REVIEW

File: docs/api/session.md [hunk 1]
```diff
@@ -348,8 +348,14 @@ win.webContents.session.setCertificateVerifyProc((request, callback) => {
 
 * `handler` Function | null
   * `webContents` [WebContents](web-contents.md) - WebContents requesting the permission.  Please note that if the request comes from a subframe you should use `requestingUrl` to check the request origin.
-  * `permission` String - Enum of 'media', 'geolocation', 'notifications', 'midiSysex',
-    'pointerLock', 'fullscreen', 'openExternal'.
+  * `permission` String - Types of permission that can be requested
```
Comment (codebytere) [inline docs/api/session.md:351]: can you add a period here for consistency?

Comment (codebytere) [reply docs/api/session.md:351]: this should also be formatted like so:
```
`permission` String - Types of permission that can be requested.
   * `media` - Allows app to access media files from a user's device with the user's consent.
   * etc
```

There's no need to redundantly specify string

```

**Comment by kritisingh1** [comment]
Updated @codebytere

File: docs/api/session.md [hunk 2]
```diff
@@ -348,8 +348,14 @@ win.webContents.session.setCertificateVerifyProc((request, callback) => {
...
-  * `permission` String - Enum of 'media', 'geolocation', 'notifications', 'midiSysex',
-    'pointerLock', 'fullscreen', 'openExternal'.
+  * `permission` String - Types of permission that can be requested.
+    * `media` - Allows app to access media files from a user's device with the user's consent.
+    * `geolocation` - Allows app to get the user's current location.
+    * `notifications` - Allows app to create notifications and display them in the user's system tray.
+    * `midiSysex` - Allows the usage of sysex messages in the `webmidi` API.
+    * `pointerLock` - Allows the app to directly interpret mouse movements as an input method.
+    * `fullscreen` - Allows application to run in fullscreen mode.
+    * `openExternal` - Allows the app to open specific application based on the url and filetype association.
```
Comment (malept) [inline docs/api/session.md:358]:
```suggestion
    * `openExternal` - Allows the app to open a specific application based on the URL and file type association.
```

File: docs/api/session.md [hunk 3]
```diff
@@ -348,8 +348,14 @@ win.webContents.session.setCertificateVerifyProc((request, callback) => {
...
   * `webContents` [WebContents](web-contents.md) - WebContents requesting the permission.  Please note that if the request comes from a subframe you should use `requestingUrl` to check the request origin.
-  * `permission` String - Enum of 'media', 'geolocation', 'notifications', 'midiSysex',
-    'pointerLock', 'fullscreen', 'openExternal'.
+  * `permission` String - Types of permission that can be requested.
+    * `media` - Allows app to access media files from a user's device with the user's consent.
+    * `geolocation` - Allows app to get the user's current location.
+    * `notifications` - Allows app to create notifications and display them in the user's system tray.
+    * `midiSysex` - Allows the usage of sysex messages in the `webmidi` API.
+    * `pointerLock` - Allows the app to directly interpret mouse movements as an input method.
+    * `fullscreen` - Allows application to run in fullscreen mode.
```
Comment (malept) [inline docs/api/session.md:357]:
```suggestion
    * `fullscreen` - Allows the app to run in fullscreen mode.
```

File: docs/api/session.md [hunk 4]
```diff
@@ -348,8 +348,14 @@ win.webContents.session.setCertificateVerifyProc((request, callback) => {
 
 * `handler` Function | null
   * `webContents` [WebContents](web-contents.md) - WebContents requesting the permission.  Please note that if the request comes from a subframe you should use `requestingUrl` to check the request origin.
-  * `permission` String - Enum of 'media', 'geolocation', 'notifications', 'midiSysex',
-    'pointerLock', 'fullscreen', 'openExternal'.
+  * `permission` String - Types of permission that can be requested.
+    * `media` - Allows app to access media files from a user's device with the user's consent.
+    * `geolocation` - Allows app to get the user's current location.
+    * `notifications` - Allows app to create notifications and display them in the user's system tray.
```
Comment (malept) [inline docs/api/session.md:354]:
```suggestion
    * `notifications` - Allows notification creation and the ability to display them in the user's system tray.
```

File: docs/api/session.md [hunk 5]
```diff
@@ -348,8 +348,14 @@ win.webContents.session.setCertificateVerifyProc((request, callback) => {
 
 * `handler` Function | null
   * `webContents` [WebContents](web-contents.md) - WebContents requesting the permission.  Please note that if the request comes from a subframe you should use `requestingUrl` to check the request origin.
-  * `permission` String - Enum of 'media', 'geolocation', 'notifications', 'midiSysex',
-    'pointerLock', 'fullscreen', 'openExternal'.
+  * `permission` String - Types of permission that can be requested.
+    * `media` - Allows app to access media files from a user's device with the user's consent.
+    * `geolocation` - Allows app to get the user's current location.
```
Comment (malept) [inline docs/api/session.md:353]:
```suggestion
    * `geolocation` - Allows access to the user's current location.
```

File: docs/api/session.md [hunk 6]
```diff
@@ -348,8 +348,14 @@ win.webContents.session.setCertificateVerifyProc((request, callback) => {
 
 * `handler` Function | null
   * `webContents` [WebContents](web-contents.md) - WebContents requesting the permission.  Please note that if the request comes from a subframe you should use `requestingUrl` to check the request origin.
-  * `permission` String - Enum of 'media', 'geolocation', 'notifications', 'midiSysex',
-    'pointerLock', 'fullscreen', 'openExternal'.
+  * `permission` String - Types of permission that can be requested.
+    * `media` - Allows app to access media files from a user's device with the user's consent.
```
Comment (malept) [inline docs/api/session.md:352]:
```suggestion
    * `media` - Allows access to media files from a device with the user's consent.
```

File: docs/api/session.md [hunk 7]
```diff
@@ -348,8 +348,14 @@ win.webContents.session.setCertificateVerifyProc((request, callback) => {
...
+    * `geolocation` - Allows app to get the user's current location.
+    * `notifications` - Allows app to create notifications and display them in the user's system tray.
+    * `midiSysex` - Allows the usage of sysex messages in the `webmidi` API.
```
Comment (malept) [inline docs/api/session.md:355]:
```suggestion
    * `midiSysex` - Allows the use of sysex messages in the `webmidi` API.
```

File: docs/api/session.md (hunk 2 above)
Comment (erickzhao) [reply docs/api/session.md:358]: I'm unclear as to what "file type association" means here. From what I understand from the original PR (#5231), this is meant for custom browser protocols.

Comment (deepak1556) [reply docs/api/session.md:358]: Not browser protocols but generally schemes that are registered by other applications they can handle with the OS, ex: `mailto`, `ms-settings:` etc, so these are schemes external to your app.

File: docs/api/session.md (hunk 6 above)
[the remark this answers is not in the document]
Comment (deepak1556) [reply docs/api/session.md:352]: Yes thats correct, its for media devices camera, microphone.

File: docs/api/session.md (hunk 3 above)
Comment (deepak1556) [reply docs/api/session.md:357]: `run in fullscreen mode` -> `enter fullscreen mode`

**Comment by deepak1556** [comment]
Additional thought, the permission types currently reads `Allows the app...` , the types by themselves don't allow anything, the api is mainly to respect the request from those permission types. So it would be better if the starting were rephrased to `Request to...`

**Comment by kritisingh1** [comment]
I've incorporated all the suggestions. Please have a look!

**Comment by kritisingh1** [comment]
I looked at the build-mac logs, but couldn't determine why is there a problem. I would be very grateful if you can help me here. Thanks!
@codebytere @erickzhao

**Pull request merged**